src/domain/repositories/ProgramsRepository.ts

```diff
@@ -1,20 +1,47 @@
-import { Async } from "domain/entities/Async";
-import { Id } from "domain/entities/Base";
-import { Pair } from "scripts/common";
+import { Enrollment, Event, TrackedEntity } from "domain/entities/TrackedEntity";
+import { CancelableResponse } from "types/d2-api";
 
 export interface ProgramsRepository {
-    closePatients(options: ClosePatientsOptions): Async<void>;
+    get(options: GetOptions): CancelableResponse<ApiGetResponse>;
```

Comment:
`CancelableResponse`. That's a specific d2-api type that should leak to the domain. In the domain, we use domain types like Future/Promise/Async. Here, `Async`.